eng/common/scripts/Update-ChangeLog.ps1

```diff
@@ -80,40 +84,44 @@ if ($ChangeLogEntries.Contains($Version))
 $PresentVersionsSorted = [AzureEngSemanticVersion]::SortVersionStrings($ChangeLogEntries.Keys)
 $LatestVersion = $PresentVersionsSorted[0]
 
+LogDebug "The latest release note entry in the changelog is for version [$($LatestVersion)]"
+
 $LatestsSorted = [AzureEngSemanticVersion]::SortVersionStrings(@($LatestVersion, $Version))
 if ($LatestsSorted[0] -ne $Version) {
-    LogWarning "Passed Version [$Version] is older than the latestversion [$LatestVersion] in the changelog. Please use a more recent version."
+    LogWarning "Version [$Version] is older than the latestversion [$LatestVersion] in the changelog. Please use a more recent version."
     exit(0)
 }
 
-if ($ReplaceLatestEntry) 
+if ($ReplaceLatestEntryTitle) 
 {
+    $newChangeLogEntry = New-ChangeLogEntry -Version $Version -Status $ReleaseStatus -Content $ChangeLogEntries[$LatestVersion].ReleaseContent
+    LogDebug "Resetting latest entry title to [$($newChangeLogEntry.ReleaseTitle)]"
     $ChangeLogEntries.Remove($LatestVersion)
-    $newChangeLogEntry = New-ChangeLogEntry -Version $Version -Status $ReleaseStatus
     if ($newChangeLogEntry) {
         $ChangeLogEntries[$Version] = $newChangeLogEntry
     }
     else {
         LogError "Failed to create new changelog entry"
+        exit 1
     }
 }
 elseif ($ChangeLogEntries.Contains($Version))
 {
-    $ChangeLogEntries[$Version].ReleaseVersion = $Version
```

Comment:
Do you know if there was any reason why ReleaseVersion wouldn't match Version? Did we used to set it when we didn't need to?

Reply:
The ReleaseVersion should be the same as the Version in the title of the Release Entry. Its just that its named ReleaseVersion in the object returned from the ChangeLog operation.

Reply:
Ok if there is no case were we need to update it then I'm fine with removing this.